src/components/ViewContracts/container.js

```diff
@@ -18,67 +25,121 @@ export default Wrapped =>
     constructor(props) {
       super(props);
       this.state = {
-        users: [],
+        employeesOptions: [],
+        selectedEmployeeId: -1,
+        selectedTeamId: -1,
         contracts: [],
       };
     }
 
-    getContracts = (selectedUserId) => {
-      getContractsByEmployeeId(selectedUserId)
+    _byEmployeeId = (selectedEmployeeId) => {
+      getContractsByEmployeeId(selectedEmployeeId).then(response => {
+        const contracts = response.status === 200 ? response.data : [];
+        this.setState({ contracts });
+      })
+        .catch(error =>
+          swal('Error'`Error finding contracts: ${error.message}`, 'error')
+        );
+    }
+
+    _byTeamId = (selectedTeamId) => {
+      getContractsByTeamId(selectedTeamId)
+        .then(response => {
+          const contracts = response.status === 200 ? response.data : [];
+          this.setState({ contracts });
+        })
+        .catch(error =>
+          swal('Error'`Error finding contracts: ${error.message}`, 'error')
+        );
+    }
+
+    _byEmployeeIdAndTeamId = (selectedEmployeeId, selectedTeamId) => {
+      getContractByEmployeeIdAndTeamId(selectedEmployeeId, selectedTeamId)
         .then(response => {
           const contracts = response.status === 200 ? response.data : [];
           this.setState({ contracts });
         })
         .catch(error =>
           swal('Error'`Error finding contracts: ${error.message}`, 'error')
         );
+    }
+
+    getContracts = (name, id) => {
+
+      const obj = {};
+      obj[name] = parseInt(id);
+
+      this.setState(obj, () => {
+
+        const { selectedEmployeeId, selectedTeamId } = this.state;
+
+        const searchByEmployees = selectedEmployeeId !== -1 && selectedTeamId === -1;
+        const searchByTeams = selectedTeamId !== -1 && selectedEmployeeId === -1;
+        const searchByBoth = selectedTeamId !== -1 && selectedEmployeeId !== -1;
+
+        if (searchByEmployees) {
```

Comment:
A small point but one to consider when on projects.

You have

```
        if (searchByEmployees) {
          this._byEmployeeId(selectedEmployeeId);
        }
        else if (searchByTeams) {
          this._byTeamId(selectedTeamId);
        }
        else if (searchByBoth) {
          this._byEmployeeIdAndTeamId(selectedEmployeeId, selectedTeamId);
        } else {
          this.setState({ contracts: [] });
        }
```

but keep it consistent with a change

 ```
       if (searchByEmployees) {
          this._byEmployeeId(selectedEmployeeId);
        } else if (searchByTeams) {
          this._byTeamId(selectedTeamId);
        } else if (searchByBoth) {
          this._byEmployeeIdAndTeamId(selectedEmployeeId, selectedTeamId);
        } else {
          this.setState({ contracts: [] });
        }
```

Again a small opinion but it just makes the file look tidier.

Reply:
If statement structure updated to look tidier.